### ISSUE TITLE
Consider using progressr for compile progress indicator (possibly others?)

### ISSUE THREAD

**Issue by mike-lawrence** [stan-dev/cmdstanr#486]
The latest release added a spinner when compiling, which is nice but adds spam when sinking output to file in quasi-interactive sessions (been playing with SBC a bunch lately). 

I really like the design motto of the [progressr package](https://github.com/HenrikBengtsson/progressr):

> The developer is responsible for providing progress updates but it's only the end user who decides if, when, and how progress should be presented. No exceptions will be allowed.

Obviously using progressr creates a dependency, but I think it might be worth it?

**Comment by rok-cesnovar** [comment]
> The latest release added a spinner when compiling,

Hmmm, not sure, but wasnt this there for a few releases already? I dont recall any recent changes on this.

I think this spinner is not that important of a feature to be worth it a new dependancy. If the spinner causes problems, lets just remove it or remove it if request  (quiet = TRUE or the like).

Does this also print if you go with quiet = TRUE? I think we just need to improve the interface wrt to quiet/verbosity levels.

**Comment by jgabry** [comment]
On Mon, Apr 19, 2021 at 5:49 AM Rok Češnovar ***@***.***>
wrote:

> The latest release added a spinner when compiling,
>
> Hmmm, not sure, but wasnt this there for a few releases already? I dont
> recall any recent changes on this.
>

Yeah that's not new, it's been there since the earliest stages of the
package.


> I think this spinner is not that important of a feature to be worth it a
> new dependancy. If the spinner causes problems, lets just remove it or
> remove it if request (quiet = TRUE or the like).
>
Agreed. I think we did it initially just so there would be some feedback to
indicate that something is happening rather than total silence during
compilation, but it's not super important. I'd be fine with removing it.

**Comment by jgabry** [comment]
> So in interactive session it's not possible to get rid of it without getting other messages! Also calling `rmarkdown::render(...)` from interactive session is considered interactive, so the generated html has hundreds of lines of

Yeah that's bad! I'd be ok with getting rid of it or with adding an option to turn it off. 

@rok-cesnovar Do you have a preference?

**Comment by jgabry** [comment]
> Also calling rmarkdown::render(...) from interactive session is considered interactive, so the generated html has hundreds of lines of

Coming back to this as I'm going through old issues. Years ago we fixed this to use `rlang::is_interactive()`, which returns FALSE while knitr is rendering, so the R Markdown/Quarto HTML spam should be gone, but we never updated this issue. 

In v1.0 there will also be a global option to turn off the spinner entirely even in real interactive sessions: https://github.com/stan-dev/cmdstanr/pull/1224
